pkg/services/folder/folderimpl/folder.go

```diff
@@ -280,7 +281,12 @@ func (s *Service) Create(ctx context.Context, cmd *folder.CreateFolderCommand) (
 	dashFolder.SetUID(trimmedUID)
 
 	user := cmd.SignedInUser
-	userID := user.UserID
+	namespaceID, userIDstr := user.GetNamespacedID()
```

Comment:
check namespace, if API key set user ID to 0 if you want to keep legacy support for API keys.